libs/common/src/makeException.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2019 - 2019 Settlers Freaks (sf-team at siedler25.org)
+//
+// This file is part of Return To The Roots.
+//
+// Return To The Roots is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Return To The Roots is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Return To The Roots. If not, see <http://www.gnu.org/licenses/>.
+
+#include "makeException.h"
+#include <boost/system/config.hpp>
+#ifdef BOOST_WINDOWS_API
+#include <windows.h>
+#elif defined(BOOST_POSIX_API)
+#include <errno.h>
```

Comment:
shouldn't it be 'cerrno'? and why not removing boost here completly and use `_WIN32` instead?

Reply:
BEcause this should also be used for cygwin. The boost define has both although we could do it manually.

Reply:
ah yes I don't know if cygwin defines _WIN32 and since we're using boost anyway ...